modules/images/image-loading-optimization/detection/detect.js

```diff
@@ -104,19 +106,36 @@ function error( ...message ) {
 /**
  * Gets element index among siblings.
  *
+ * @todo Eliminate this in favor of doing all breadcrumb generation exclusively on the server.
+ *
  * @param {Element} element Element.
  * @return {number} Index.
  */
 function getElementIndex( element ) {
 	if ( ! element.parentElement ) {
 		return 0;
 	}
-	return [ ...element.parentElement.children ].indexOf( element );
+	const children = [ ...element.parentElement.children ];
+	let index = children.indexOf( element );
+	if ( children.includes( document.getElementById( adminBarId ) ) ) {
+		// TODO: Should detection just be turned off when is_user_logged_in()?
```

Comment:
> Should detection just be turned off when is_user_logged_in()?

probably, the whole point of this exercise is to gather real user metrics.

Reply:
What about sites that require users to be logged in to see certain content or comment?

Reply:
Yeah, good point. I'm removing this todo.

```suggestion
```